File: civitasruby/lib/casilla.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.



module Civitas
class Casilla
  attr_reader :nombre,:tituloPropiedad
  
  
  
  def initialize(nombre)
      @nombre= nombre
  end
  
  def recibe_jugador(actual,todos)
      informe(actual,todos)
  end
  
  def informe(actual,todos)
    Diario.instance.ocurre_evento("El jugador "+todos[actual].nombre+" ha caido en la casilla "+to_string)
  end
  
  def jugador_correcto(actual,todos)
    return actual<=todos.size
  end
  

  
  def to_string
    cadena="Casilla: #{@nombre} \n"
    cadena+="\n"
    return cadena
  end
  
  
  
  
  
  
  private :informe
  end 
end

File: civitasjava/src/Civitas/Casilla.java
package civitas;
import java.util.ArrayList;

public class Casilla {
    private String nombre;
    
    Casilla(String nombre){
        this.nombre = nombre;
    }
    
    public String getNombre(){
        return nombre;
    }
    
    public void informe(int actual, ArrayList<Jugador> todos){
        Diario.getInstance().ocurreEvento("El jugador "+
                todos.get(actual).getNombre()+ "ha caido en la casilla " 
                +toString());
    }
    
    @Override
    public String toString(){
        return "Casilla_Descanso{ " + "nombre: " +nombre + "}";
    }
    
    public boolean jugadorCorrecto(int actual, ArrayList<Jugador> todos){
        return actual <= todos.size();
    }
    
    public void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            informe(actual,todos);
        }
    }

}

